xgfield/fieldsky.py

```diff
@@ -0,0 +1,105 @@
+def _get_websky_displacements():
+    # Websky displacement fields
+    try:
+        path2disp = os.environ['LPT_DISPLACEMENTS_PATH']
+    except:
+        path2disp = '/Users/shamik/Documents/Work/websky_datacube/'
+
+    backend.print2log(log, f"Path to displacement files set to {path2disp}", level='usky_info')
+
+    if grid_nside == 768:
+        sxfile = path2disp+'sx1_7700Mpc_n6144_nb30_nt16_no768'
+        syfile = path2disp+'sy1_7700Mpc_n6144_nb30_nt16_no768'
+        szfile = path2disp+'sz1_7700Mpc_n6144_nb30_nt16_no768'
+    else:
+        sxfile = path2disp+'sx1_7700Mpc_n6144_nb30_nt16'
+        syfile = path2disp+'sy1_7700Mpc_n6144_nb30_nt16'
+        szfile = path2disp+'sz1_7700Mpc_n6144_nb30_nt16'
+
+    displacements = {}
+    displacements['type'] = 'filelist'
+    displacements['data'] = [sxfile, syfile, szfile]
+
+    return displacements
+
+class FieldSky:
+    '''FieldSky'''
+    def __init__(self, **kwargs):
+
+        import xgfield.defaults as fd
+        self.ID    = kwargs.get(    'ID',fd.ID)
+        self.N     = kwargs.get(     'N',fd.N)
+        self.Lbox  = kwargs.get(  'Lbox',fd.Lbox)
+        self.Nside = kwargs.get( 'Nside',fd.Nside)
+        self.input = kwargs.get( 'input',fd.input)
+        self.nproc = kwargs.get( 'nproc',1)
```

Comment:
I am not sure about creating a FieldSky object that needs the MPI params as input. Typically my hope was to separate the user interfacing objects and the backend.

Reply:
@1cosmologist thank, I also hope to keep MPI params in a single object i.e. the backend, but I think trying to fully refactor that in this PR will delay using the other features already working. How about spinning this off into a separate issue so we can track progress there?